src/components/tools/use-image-content-url.ts

```diff
@@ -8,14 +8,10 @@ type OnUrlChangeFn = (url: string, filename?: string, context?: IDocumentContext
 
 export function useImageContentUrl(content: ImageContentModelType, onUrlChange: OnUrlChangeFn) {
   const context = useContext(DocumentContextReact);
-  const syncedChanges = useRef(0);
   useEffect(() => {
     const dispose = autorun(() => {
-      if (content.changeCount > syncedChanges.current) {
```

Comment:
When the changes approach is removed, I'd think that this autorun could be removed, and anything needing the url from the content would just read it. If a component needs to update when url changes it should be marked as `observable`.

Reply:
What @scytacki says is true, that one way to handle this would be to eliminate the `autorun` and the `useEffect` entirely and rely on clients to observe the model directly. In the short term, it may be useful to minimize effects on downstream clients in which case there could still be an `autorun` or `reaction` here that just observed the `url` and called the client's callback function whenever it changes. It depends in part on how many clients there are and how hard it would be to change them.

Reply:
The goal here is to have the provided `onUrlChange` function be called when the `url` changes. As written this will call `onUrlChange` when the `content` changes but not when the `url` changes. I think what you probably want here is something like:
```
    const dispose = reaction(() => ({ url: content.url, filename: content.filename }), ({ url, filename }) => url && onUrlChange(url, filename, context));
    return () => dispose();
```

Reply:
If this is working I'm fine leaving it for this PR. But for the record:

`useImageContentUrl` seems overly complex. It is really just a way for the ImageTool to get the documentContext and monitor changes in the url of the image content model. The ImageTool could do both of those things directly without having pass a handler through the ImageComponent and then into useImageContentUrl.  The ImageComponent doesn't do anything with this information so it is weird that is the man in the middle. I suspect it is just because it is a functional component so it makes it easier to call `useContext`.

The change in the url seems mainly needed so the ImageTool `isLoading` state property can be set true. This also seems strange. I'd think the component would just want to wait for a valid displayUrl on the image entry in the cache. As soon as it gets that, it is no longer loading. The component shouldn't really care if the source/original/content url is getting switched around.